src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,382 @@
+
+                if (!file_exists($directory)) {
+                    throw new InvalidArgumentException(sprintf('Cache directory does not exist and cannot be created: %s', $directory));
+                }
+            }
+
+            if (!is_writable($directory)) {
+                throw new InvalidArgumentException(sprintf('Cache directory is not writable: %s', $directory));
+            }
+        }
+
+        $this->fallbackPool->clear();
+
+        $dump = <<<EOF
+<?php
+
+// This file has been auto-generated by the Symfony Cache Component.
+
+return array(
+
+
+EOF;
+
+        foreach ($values as $key => $value) {
+            $fallbackItem = $this->fallbackPool->getItem($key);
+
+            if (null === $value || is_object($value)) {
+                try {
+                    $value = serialize($value);
+                } catch (\Exception $e) {
+                    throw new InvalidArgumentException(sprintf('Cache key "%s has non-serializable %s value', $key, get_class($value)), 0, $e);
+                }
+            } elseif (is_array($value)) {
+                try {
+                    $serialized = serialize($value);
+                    $unserialized = unserialize($serialized);
+                } catch (\Exception $e) {
+                    throw new InvalidArgumentException(sprintf('Cache key "%s has non-serializable array value', $key), 0, $e);
+                }
+                // Store arrays serialized if they contain any objects or references
+                if ($unserialized !== $value || (false !== strpos($serialized, ';R:') && preg_match('/;R:[1-9]/', $serialized))) {
+                    $value = $serialized;
+                }
+            } elseif (is_string($value)) {
+                // Serialize strings if they could be confused with serialized objects or arrays
+                if ('N;' === $value || (isset($value[2]) && ':' === $value[1])) {
+                    $value = serialize($value);
+                }
+            } elseif (!is_scalar($value)) {
+                throw new InvalidArgumentException(sprintf('Cache key "%s has non-serializable %s value', $key, gettype($value)));
+            }
+
+            $this->fallbackPool->saveDeferred($fallbackItem->set($value));
+            $dump .= var_export($key, true).' => '.var_export($value, true).",\n";
+        }
+
+        $this->fallbackPool->commit();
+        $dump .= "\n);\n";
+        $dump = str_replace("' . \"\\0\" . '", "\0", $dump);
+
+        $tmpFile = tempnam(dirname($this->file), 'sf_opcache_adapter');
+
+        file_put_contents($tmpFile, $dump);
+        unset($serialized, $unserialized, $value, $fallbackItem, $dump);
+
+        if ('\\' === DIRECTORY_SEPARATOR) {
+            // Windows does not support overwrite during rename
+            @unlink($this->file);
+            @rename($tmpFile, $this->file);
+        } else {
+            rename($tmpFile, $this->file);
+        }
+
+        $this->values = (include $this->file) ?: array();
```

Comment:
At this moment, we want to be warned if there is an issue with the inclusion of the file. We are at the storage moment: if the warmup failed, we want to be alerted. However, we don't want to stop the warm-up as we may be in production and other warmers may work. So this is a compromise: a notice to know there was a problem but without blocking the rest of the process.